packages/core/src/RenderingEngine/Viewport.ts

```diff
@@ -682,6 +707,114 @@ class Viewport implements IViewport {
     return true;
   }
 
+  public resetPanOffset(widthWorld?: number, heightWorld?: number) {
+    if (widthWorld && heightWorld) {
+      const camera = this.getCamera();
+      const activeCamera = this.getVtkActiveCamera();
+      const { viewPlaneNormal, focalPoint, position, viewUp } = camera;
+      const distanceVec = subtractVec3(position, focalPoint);
+
+      const { initialCenter = [0.5, 0.5] } = this.options;
+      const { canvasCenter = initialCenter } = this.options;
+
+      if (!focalPoint || focalPoint[0] === null) return;
+
+      const worldSizeError = 1.15;
+
+      // The focal point is currently the center of the image
+      // It needs to be adjusted by the sum of:
+      // vector between center and initialCenter in image size coordinates
+      // vector between the canvas center and the specific canvasCenter
+      if (initialCenter[0] !== 0.5) {
+        const delta = 0.5 - initialCenter[0];
+        const viewRight = vec3.create();
+        vec3.cross(viewRight, viewPlaneNormal, viewUp);
+        vec3.scaleAndAdd(
+          focalPoint,
+          focalPoint,
+          viewRight,
+          delta * widthWorld * worldSizeError
+        );
+      }
+      if (initialCenter[1] !== 0.5) {
+        const delta = 0.5 - initialCenter[1];
+        const viewUp = activeCamera.getViewUp() as Point3;
+        vec3.scaleAndAdd(
+          focalPoint,
+          focalPoint,
+          viewUp,
+          delta * heightWorld * worldSizeError
+        );
+      }
+      const topLeft = this.canvasToWorld([0, 0]);
+      if (canvasCenter[0] !== 0.5) {
+        const delta = 0.5 - canvasCenter[0];
+        const right = this.canvasToWorld([this.sWidth, 0]);
+        const vector = subtractVec3(right, topLeft);
+        vec3.scaleAndAdd(focalPoint, focalPoint, vector, delta);
+      }
+      if (canvasCenter[1] !== 0.5) {
+        const delta = 0.5 - canvasCenter[1];
+        const bottom = this.canvasToWorld([0, this.sHeight]);
+        const vector = subtractVec3(bottom, topLeft);
+        vec3.scaleAndAdd(focalPoint, focalPoint, vector, delta);
+      }
+      activeCamera.setFocalPoint(focalPoint[0], focalPoint[1], focalPoint[2]);
+      activeCamera.setPosition(
+        focalPoint[0] + distanceVec[0],
+        focalPoint[1] + distanceVec[1],
+        focalPoint[2] + distanceVec[2]
+      );
+    }
+
+    this.panOffset = [0, 0];
+    this.panOffset = this.getCanvasPan();
+  }
+
+  // The initial canvas pan of 0,0 means
```

Comment:
Explain that this is just a function of the current camera position that returns values in terms of the initial position of the canvas.